data_processors/pipeline/domain/workflow.py

```diff
@@ -153,20 +153,17 @@ def get_ssm_key_input(self):
     def construct_workflow_name(self, **kwargs):
         # pattern: [AUTOMATION_PREFIX]__[WORKFLOW_TYPE]__[WORKFLOW_SPECIFIC_PART]__[UTC_TIMESTAMP]
         utc_now_ts = int(datetime.utcnow().replace(tzinfo=timezone.utc).timestamp())
+        portal_uuid = kwargs.get('portal_uuid', "")
         # Primary analysis
         if self.type == WorkflowType.BCL_CONVERT:
             seq_name = kwargs['seq_name']
             seq_run_id = kwargs['seq_run_id']
-            return f"{WorkflowHelper.prefix}__{self.type.value}__{seq_name}__{seq_run_id}__{utc_now_ts}"
+            return f"{WorkflowHelper.prefix}__{self.type.value}__{seq_name}__{seq_run_id}__{portal_uuid}__{utc_now_ts}"
```

Comment:
ditto consistent `portal_run_id` naming; instead of `portal_uuid`

Reply:
I don't mind it there. The portal uuid has no date time information. We probably don't need the seq_run_id though.

Reply:
No strong opinion either... it's just that it does not seem to fulfil any particular purpose.
The date/time are already covered by the workflow metadata itself. The date is also covered in the `portal_run_id`, which also makes the name unique.
Agree with removing the `seq_run_id`, that is unlikely to change for any given `seq_name` (and even if it does, we can trace that internally if need be).

Reply:
Btw: any particular reason why we use `-` instead of `_` in the `portal_run_id`?
That's the only component that results in dashes in folder names (I think). Also it could be shortened to `yyyymmdd_uid` or even `yymmdd_uid` to make it shorter while preserving the distinct `__` separator in the workflow name.

Reply:
> Btw: any particular reason why we use - instead of _ in the portal_run_id


As in between the date time stamp and the run uuid in the gds path? We can change it to `_` or `__` if preferred?

> Also it could be shortened to yyyymmdd_uid or even yymmdd_uid to make it shorter while preserving the distinct __ separator in the workflow name.

Happy to truncate to `yyyymmdd_uid`.

Reply:
>> Btw: any particular reason why we use - instead of _ in the portal_run_id

> As in between the date time stamp and the run uuid in the gds path?

Within the run id: 2021-11-10--a1b2c3e4

> Happy to truncate to yyyymmdd_uid

Perfect!

Reply:
`portal_run_id` is now defined as yyyymmdd + <8 digit UID>, e.g. `20211105a1b2c3e4`
 Happy to hear opinions...

Reply:
I prefer without underscore form (perfect form for ID). But does it loose sort and, may be readability a tad? `ls` seems to be ok. Finder (and in my PyCharm) it seem to be confused numerical from uuid chunk. I'm trying like:

```python
"""
Usage:
    cd tmp
    python pid.py
    rm -rf 20*
"""
import os
import uuid

if __name__ == '__main__':
    for i in range(1, 31):
        datestamp = f"201201{i:02}"
        gen_uuid = lambda: f"{str(uuid.uuid4())[:8]}"
        if i in [1, 3, 5, 10, 11]:
            os.mkdir(f"{datestamp}{gen_uuid()}")  # create additional within the same datestamp
        os.mkdir(f"{datestamp}{gen_uuid()}")
```

![image](https://user-images.githubusercontent.com/4586509/141387000-515c6138-92a3-4aea-a8e2-f22888384434.png)

Reply:
This folder will be a subfolder of `subject_id/workflow-type` so unless we have multiple workflows running on the same day for a given subject with a given workflow type, I don't think it's going to be too much of an issue